pkg/api/v1alpha1/provider_spec.go

```diff
@@ -195,6 +200,17 @@ type GCPDisk struct {
 	// the disks.setLabels method. This field is only applicable for
 	// persistent disks.
 	Labels map[string]string `json:"labels"`
+
+	// ProvisionedIops of disk to create.
+	// Only for use with disks of type pd-extreme and hyperdisk-extreme.
+	// The IOPS must be specified within defined limits
+	ProvisionedIops int64 `json:"provisionedIops"`
```

Comment:
This is optional and therefore please add `omitempty` to the json tag.
If one does not set this value then it is defaulted to 0. Will GCP default it to a valid value?
If you read the doc string in their code you will find this:
```
	// ProvisionedIops: Indicates how many IOPS to provision for the disk.
	// This sets the number of I/O operations per second that the disk can
	// handle. Values must be between 10,000 and 120,000. For more details,
	// see the Extreme persistent disk documentation.
	ProvisionedIops int64 `json:"provisionedIops,omitempty,string"`

	// ProvisionedThroughput: Indicates how much throughput to provision for
	// the disk. This sets the number of throughput mb per second that the
	// disk can handle. Values must be between 1 and 7,124.
	ProvisionedThroughput int64 `json:"provisionedThroughput,omitempty,string"`
```
This means 0 is not a valid value for any of the above. Since their documentation does not mention anything regarding default value can you test with explicitly setting these fields to 0?

Reply:
@unmarshall but the GCP client also has the same fields at non-pointers and we were already initialising the enclosing struct, ergo we were already using the `0` value anyway.

In reality, the GCP client will not write to the request any zero value unless instructed to do so explicitly (you need to write the field name to a differnt field for that). In our case, the 0 would basically be omitted from the request resulting in the current behavior